onnxruntime/contrib_ops/cpu/bert/attention_helper.h

```diff
@@ -41,22 +41,23 @@ void ComputeAttentionSoftmaxInplace(T* score, int N, int D, ThreadPool* tp) {
         sum += x[i];
       }
 
-      if (sum == 0) {
-        for (int i = 0; i < D; i++) {
-          y[i] = 1.0f / (float)D;
-        }
-      } else {
+      // `sum` (which we are dividing by below) will never be 0 as long as there is atleast one element in the sequence
+      // (i.e.) D > 0 (sequence_length > 0). If we reach this line of code, we have already ensured that
+      // sequence_length > 0, when we check if the input tensor is non-empty
+      // TODO: Account for cases where mask == 0 and sequence_length > 0 when "masked" softmax is eventually implemented
+      // as sum will be zero in the case
+      for (int i = 0; i < D; i++) {
+        y[i] = x[i] / (float)sum;
         for (int i = 0; i < D; i++) {
           y[i] = x[i] / (float)sum;
         }
       }
-    }
-  });
+    });
 }
 
 template <>
 inline void ComputeAttentionSoftmaxInplace(float* score, int N, int D, ThreadPool* tp) {
-  MlasComputeSoftmax(score, score, N, D, false, tp);
+    MlasComputeSoftmax(score, score, N, D, false, tp);
```

Comment:
not clear why the spacing changed here, this is from clang-format? the below changes also look off?

Reply:
should be okay now -all of it was just the artifact of the merge problem